crates/language/src/php.rs

```diff
@@ -0,0 +1,79 @@
+use std::sync::OnceLock;
+
+use crate::language::{fields_for_nodes, Field, Language, SortId, TSLanguage};
+
+static NODE_TYPES_STRING: &str = include_str!("../../../resources/node-types/php-node-types.json");
+static NODE_TYPES: OnceLock<Vec<Vec<Field>>> = OnceLock::new();
+static LANGUAGE: OnceLock<TSLanguage> = OnceLock::new();
+
+#[cfg(not(feature = "builtin-parser"))]
+fn language() -> TSLanguage {
+    unimplemented!(
+        "tree-sitter parser must be initialized before use when [builtin-parser] is off."
+    )
+}
+#[cfg(feature = "builtin-parser")]
+fn language() -> TSLanguage {
+    tree_sitter_php::language().into()
+}
+
+#[derive(Debug, Clone)]
+pub struct Php {
+    node_types: &'static [Vec<Field>],
+    metavariable_sort: SortId,
+    language: &'static TSLanguage,
+}
+
+impl Php {
+    pub(crate) fn new(lang: Option<TSLanguage>) -> Self {
+        let language = LANGUAGE.get_or_init(|| lang.unwrap_or_else(language));
+        let node_types = NODE_TYPES.get_or_init(|| fields_for_nodes(language, NODE_TYPES_STRING));
+        let metavariable_sort = language.id_for_node_kind("grit_metavariable", true);
+        Self {
+            node_types,
+            metavariable_sort,
+            language,
+        }
+    }
+    pub(crate) fn is_initialized() -> bool {
+        LANGUAGE.get().is_some()
+    }
+}
+
+impl Language for Php {
```

Comment:
override the metavariable prefix function as well, as `$` is already used to specify variables in php. I'm not sure what the correct what would be a good substitute maybe `#` would be good to get input from someone who does more work in php